### ISSUE TITLE
\prime and \pprime phase for minute and second (angles)

### ISSUE THREAD

**Issue by ngam** [PainterQubits/Unitful.jl#522]
https://en.wikipedia.org/wiki/International_System_of_Units#Non-SI_units_accepted_for_use_with_SI

Julia does support `\prime` and `\pprime` symbols, i.e. `’` and `”` respectively. However, it only supports them as associated with other quantities it seems? For example, 

typing `a\prime[tab] = 5`:

```
julia> a′ = 5
5
```

typing `a’ = 5` (directly):

```
julia> a’ = 5
ERROR: syntax: invalid character "’" near column 2
Stacktrace:
 [1] top-level scope
   @ none:1
```

I would assume for these two units to be adopted in Unitful, one would need to convince upstream Julia to add the `\prime` and `\pprime` characters fully like for example the `\degree` character:

typing `a\degree[tab] = 5`:

```
julia> a° = 5
5
```

typing `a° = 5` (directly):

```
julia> a° = 5
5
```

For completion, what would the process be like to adopt these two units and is this something the Uniftul.jl maintainers would be interested in?

Btw, thanks for the great work! I tried all symbols and variants in that Wikipedia page and these two units were the only missing!

**Comment by sostock** [comment]
> typing `a’ = 5` (directly):
> 
> ```
> julia> a’ = 5
> ERROR: syntax: invalid character "’" near column 2
> ```

This doesn’t work because the symbol you typed isn’t a *prime* (U+2032), but a *right single quotation mark* (U+2019). If you actually type `a′ = 5` (with a prime) it works.

However, in order to be able to use `′` (*prime*, U+2032, produced by `\prime`) and `″` (*double prime*, U+2033, produced by `\pprime`) as units, they would have to be allowed as single-character identifiers, which currently isn’t the case:
```julia
julia> ′ = 5
ERROR: syntax: invalid character "′" near column 1
[...]

julia> Base.isidentifier("′")
false
```
They are not allowed as single-character identifiers because they are in the Unicode category “Punctuation”.

The modifiers `ʹ` (*modifier letter prime*, U+02B9) and `ʺ` (*modifier letter double prime*, U+02BA) *are* allowed as identifiers, since they are in the „Letter“ category. But these aren’t the characters that are produced by `\prime` and `\pprime`. We *could* add those as units, but I’m not in favour of that, because I think it would be confusing (How do you type them? Why aren’t they the symbols produced by `\prime` and `\pprime`?).

Another point to consider: `′` and `″` as unit symbols are ambiguous: the former is commonly used for both minutes *and* feet and the latter is commonly used for both seconds *and* inches.

**Comment by ngam** [comment]
> This doesn’t work because the symbol you typed isn’t a _prime_ (U+2032), but a _right single quotation mark_ (U+2019). If you actually type `a′ = 5` (with a prime) it works.

Wow! Good catch!

> We _could_ add those as units, but I’m not in favour of that, because I think it would be confusing (How do you type them? Why aren’t they the symbols produced by `\prime` and `\pprime`?).

Yes, I agree with you here now; even without adding them as units, it's already confusing!

> Another point to consider: `′` and `″` as unit symbols are ambiguous: the former is commonly used for both minutes _and_ feet and the latter is commonly used for both seconds _and_ inches.

I see. Out of curiosity, do you follow specific guidelines (e.g. NIST) or do you also take into account common usage?

Let me reiterate that the only motivation for me raising this: These two were the only missing "units" from that Wikipedia page. To be completely honest, I was surprised to learn that they were used formally like this; I always thought these were more commonly used for inches and feet... 

Thank you for the thorough response! I am going to close this, but feel free to reopen it if you want it to be visible.

**Comment by ngam** [comment]
> > typing `a’ = 5` (directly):
> > ```
> > julia> a’ = 5
> > ERROR: syntax: invalid character "’" near column 2
> > ```
> 
> This doesn’t work because the symbol you typed isn’t a _prime_ (U+2032), but a _right single quotation mark_ (U+2019). If you actually type `a′ = 5` (with a prime) it works.
> 

Just a note on this in case someone comes here: I have no idea why this doesn't work on macOS. I tried using the shortcut cmd+shift+]=’ as well as the Unicode Hex Input keyboard input with U+2032 and I had the same error. Copying the resulting character from the \prime tab completion does work. So this likely means something is wrong in the implementation of this specifically in macOS... but I don't really know. Out of curiosity, @sostock, how did you type it directly and which platform?

**Comment by sostock** [comment]
Sorry, I didn’t type it directly, I just copy-pasted it (on linux). Can you verify that the Unicode Hex Input keyboard actually creates the character U+2032 (by copying the resulting `a′ = 5` into something like unicodemap.org)?

**Comment by sostock** [comment]
I just tried it under Windows 11: typing `a′ = 5` works (`′` is typed by holding `Alt` and typing 2032 on the numpad. This requires the registry key EnableHexNumpad = 1 in HKEY_CURRENT_USER\Control Panel\Input Method).

**Comment by ngam** [comment]
Hi @sostock, thanks for your patience. I have indeed tried unicodemap.org and now it works exactly as you say. When I manage to produce the correct prime in unicode, it does work in julia correctly.

I must've been doing something wrong somewhere; I did notice that sometimes the hex input keyboard would fail and not respond, so that might be it --- or more likely, I was doing something blindly and I misspoke earlier. Sorry!

For reference, on macOS:
- adding Unicode Hex Input keyboard by going to: System Preferences > Keyboard > Input Sources > + button (lower left) > Unicode Hex Input (under Others) > add
- upon selection of Unicode Hex Input keyboard, `′` is produced by holding Option and Shift and typing 2032
